### ISSUE TITLE
Support for adding a frontend ip configuration to an existing load balancer

### ISSUE THREAD

**Issue by andre-ss6** [terraform-providers/terraform-provider-azurerm#10846]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Today it seems you can only specify frontend ip configurations inside a `azurerm_lb` resource, and not as a resource by itself. This is problematic, since it makes it impossible to add a frontend ip configuration to an existing LB.

I've tried simply doing this:

```tf
resource "azurerm_lb" "hmg-loadbalancer" { # existing load balancer
  name                = "hmg-loadbalancer"
  location            = data.azurerm_resource_group.resource_group.location
  resource_group_name = data.azurerm_resource_group.resource_group.name
  sku                 = "Basic"

  frontend_ip_configuration { # new frontend ip
    name                 = "msging-storage-httptransport"
    public_ip_address_id = azurerm_public_ip.msging-storage-httptransport-ip.id
  }
}
```

And although it understands that it needs to update the LB in-place, it also plans to remove all other frontend ips currently in the LB not specified above.

### New or Affected Resource(s)

* azurerm_lb

**Comment by favoretti** [comment]
Hi @andre-ss6, thank you for reporting this. Terraform is declarative by nature, so it calculates final state of the resource. To achieve what you want you'd need to define all of the frontend IPs in your terraform code, including the one you want to add, import the resource and then run a plan. Mixing manual config with partial terraform config is IMO seldom a good idea.

**Comment by andre-ss6** [comment]
> Hi @andre-ss6, thank you for reporting this. Terraform is declarative by nature, so it calculates final state of the resource. To achieve what you want you'd need to define all of the frontend IPs in your terraform code, including the one you want to add, import the resource and then run a plan. Mixing manual config with partial terraform config is IMO seldom a good idea.

@favoretti Yep. That's why I'm asking for a way to add a frontend ip by itself.
The code shown above was only representative of what I tried.

**Comment by andre-ss6** [comment]
Also, I just noticed: having frontend ip configurations as part of the `azurerm_lb` resource, instead of a resource by themselves, leads to some deceptive terraform plans. For instance, the one generated by the code I'm running shows 4 to add, 1 to change, 0 to delete. One could see that and think "yea, that's expected" and confirm the apply. However, what's really happening is that inside that "1 change", terraform is going to delete tens of frontend ip configurations.

**Comment by jeffreybartosiewicz** [comment]
This is a needed change.  We have a situation where we have created an azure LB in our hub and many different teams need to create frontend ip addresses assigned to this one LB.  There is currently no way for them to create a frontend ip without creating their own LB.  Configurating an LB manually through the GUI (or through azure cli or PowerShell) you are able to add new front end IPs one at a time to an existing LB so one would assume that Terraform should follow suit.

**Comment by Taha-cmd** [comment]
Any update on this? Would be nice to know whether upstream or related issues are preventing this change from being implemented or it has just not been prioritized yet.

**Comment by bcreynolds** [comment]
We also have this need. In our situation, we have one load balancer for each Azure region that's shared by multiple clients. Each client is provisioned individually with Terraform, and we want to manage each client's frontend IP configuration with the rest of their config.

**Comment by rogerioefonseca** [comment]
That is indeed very important and needed.
When creating an AKS via terraform the ARM creates the MR_* resource group and there inside it creates the Loadbalancer for the cluster, and then we cannot create an extra Frontend IP because there is no specific resource for it.
That means that we need to document a manual process informing that we need to add new Ip's via other way (CLI, GUI), which is frustrating.
